### ISSUE TITLE
Fixed Dependencies

### ISSUE THREAD

**Issue by guspower** [alfiedotwtf/travelling_salesman#1]
Hey Alfie,

Just a quick issue about the fixed dependencies - it would be useful to allow at least semver updates to the dependencies if possible (e.g. https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements). 

I use `cargo outdated` (https://github.com/kbknapp/cargo-outdated) regularly and have to exclude the module that depends on `travelling_salesman` as I get:

```
error: failed to select a version for `time`.
    ... required by package `travelling_salesman v1.1.19`
    ... which satisfies dependency `travelling_salesman = "^1.1.19"` of package `address v0.1.0 (/tmp/cargo-outdatedQa4yBA)`
versions that meet the requirements `=0.3.5` are: 0.3.5

all possible versions conflict with previously selected packages.

  previously selected package `time v0.3.7`
    ... which satisfies dependency `time = "^0.3.7"` of package `address v0.1.0 (/tmp/cargo-outdatedQa4yBA)`

failed to select a version for `time` which could resolve this conflict
```

Happy to submit and/or test a PR if it's helpful!

G

**Comment by alfiedotwtf** [comment]
Hey. Sorry about that. The reason why I force exact was that I kept getting burned by `ramp` in my `is_prime` crate.

But sure, I'm happy for the change, I don't want to inconvenience you. If you're able to do a PR, that would be great, otherwise I can get around to it in the next few days.

Thanks!

**Comment by guspower** [comment]
OK great - I'll put thru a small PR for `metaheuristics` and then another on `travelling_salesman` once that's published.

**Comment by alfiedotwtf** [comment]
Fixed with #2 :)
